llvm/lib/Target/AMDGPU/SIISelLowering.cpp

```diff
@@ -4822,6 +4822,111 @@ static MachineBasicBlock *lowerWaveReduce(MachineInstr &MI,
   return RetBB;
 }
 
+static MachineBasicBlock *lowerPseudoLaneOp(MachineInstr &MI,
```

Comment:
You should try to do this before selection. Doing it after just adds a lot of complexity and interferes with combine opportunities

Reply:
@arsenm, would "PreISelIntrinsicLowering" be a proper place for this ?

Reply:
No, that's a generic pass. I would directly handle this in the legalizer, in SIISelLowering and AMDGPULegalizer